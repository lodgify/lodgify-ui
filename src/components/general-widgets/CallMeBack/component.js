import React from 'react';
import PropTypes from 'prop-types';

import {
  CALL_ME_BACK,
  DATE,
  EMAIL,
  NAME,
  NOTES,
  PHONE,
  PROPERTY,
  SEND,
  TIME,
} from 'utils/default-strings';
import { Dropdown } from 'inputs/Dropdown';
import { Form } from 'collections/Form';
import { ICON_NAMES } from 'elements/Icon';
import { InputGroup } from 'collections/InputGroup';
import { PhoneInput } from 'inputs/PhoneInput';
import { SingleDatePicker } from 'inputs/SingleDatePicker';
import { TextArea } from 'inputs/TextArea';
import { TextInput } from 'inputs/TextInput';

/**
 * The standard widget for a user to request a call back.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  dateInputPlaceholder,
  emailInputLabel,
  errorMessage,
  headingText,
  nameInputLabel,
  notesInputLabel,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  submitButtonText,
  successMessage,
  timeInputLabel,
  timeOptions,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <SingleDatePicker name="date" placeholderText={dateInputPlaceholder} />
      <Dropdown
        icon={ICON_NAMES.CLOCK}
        label={timeInputLabel}
        name="time"
        options={timeOptions}
      />
    </InputGroup>
    <Dropdown
      label={propertyInputLabel}
      name="property"
      options={propertyOptions}
    />
    <TextArea label={notesInputLabel} name="notes" />
  </Form>
);

Component.displayName = 'CallMeBack';

Component.defaultProps = {
  dateInputPlaceholder: DATE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  headingText: CALL_ME_BACK,
  nameInputLabel: NAME,
  notesInputLabel: NOTES,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  submitButtonText: SEND,
  successMessage: '',
  timeInputLabel: TIME,
  validation: {},
};

Component.propTypes = {
  /** The placeholder for the date input. */
  dateInputPlaceholder: PropTypes.string,
  /** The label for the email input. */
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input. */
  nameInputLabel: PropTypes.string,
  /** The label for the notes input. */
  notesInputLabel: PropTypes.string,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input */
  phoneInputLabel: PropTypes.string,
  /** The label for the property input. */
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** The form submit button text */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** The label for the time input. */
  timeInputLabel: PropTypes.string,
  /** The options which the user can select for the time field. */
  timeOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    date: PropTypes.object,
    email: PropTypes.object,
    name: PropTypes.object,
    notes: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    time: PropTypes.object,
  }),
};
