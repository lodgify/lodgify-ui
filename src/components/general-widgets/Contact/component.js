import React from 'react';
import PropTypes from 'prop-types';

import { Form } from 'collections/Form';
import { InputGroup } from 'collections/InputGroup';
import { TextInput } from 'inputs/TextInput';
import { PhoneInput } from 'inputs/PhoneInput';
import { NumberInput } from 'inputs/NumberInput';
import { DateRangePicker } from 'inputs/DateRangePicker';
import { TextArea } from 'inputs/TextArea';
import { CaptchaInput } from 'inputs/CaptchaInput';
import { Dropdown } from 'inputs/Dropdown';
import {
  ARRIVAL,
  CONTACT,
  COMMENTS,
  DEPARTURE,
  EMAIL,
  GUESTS,
  NAME,
  PHONE,
  PROPERTY,
  ROOM,
  SECURITY_CODE,
  SEND,
} from 'utils/default-strings';

/**
 * The standard widget for a user contact an owner.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  arrivalDateInputLabel,
  captchaInputImage,
  captchaInputLabel,
  commentsInputLabel,
  dateRangePickerLocaleCode,
  departureDateInputLabel,
  emailInputLabel,
  errorMessage,
  guestsInputLabel,
  headingText,
  nameInputLabel,
  onChangeProperty,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  roomInputLabel,
  roomOptions,
  submitButtonText,
  successMessage,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <DateRangePicker
        endDatePlaceholderText={departureDateInputLabel}
        localeCode={dateRangePickerLocaleCode}
        name="dates"
        startDatePlaceholderText={arrivalDateInputLabel}
        width="eight"
      />
      <NumberInput
        label={guestsInputLabel}
        min={1}
        name="guests"
        width="four"
      />
    </InputGroup>
    <TextArea label={commentsInputLabel} name="comments" />
    {(roomOptions || propertyOptions) && (
      <InputGroup>
        {propertyOptions && (
          <Dropdown
            label={propertyInputLabel}
            name="property"
            onChange={onChangeProperty}
            options={propertyOptions}
          />
        )}
        {roomOptions && (
          <Dropdown label={roomInputLabel} name="room" options={roomOptions} />
        )}
      </InputGroup>
    )}
    <CaptchaInput
      image={captchaInputImage}
      label={captchaInputLabel}
      name="captcha"
    />
  </Form>
);

Component.displayName = 'Contact';

Component.defaultProps = {
  arrivalDateInputLabel: ARRIVAL,
  captchaInputLabel: SECURITY_CODE,
  commentsInputLabel: COMMENTS,
  dateRangePickerLocaleCode: undefined,
  departureDateInputLabel: DEPARTURE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  guestsInputLabel: GUESTS,
  headingText: CONTACT,
  nameInputLabel: NAME,
  onChangeProperty: Function.prototype,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  propertyOptions: null,
  roomInputLabel: ROOM,
  roomOptions: null,
  submitButtonText: SEND,
  successMessage: '',
  validation: {},
};

Component.propTypes = {
  /** The label for the arrival date input.*/
  arrivalDateInputLabel: PropTypes.string,
  /** The source url for the image to display. */
  captchaInputImage: PropTypes.string.isRequired,
  /** The label for the captcha input.*/
  captchaInputLabel: PropTypes.string,
  /** The label for the comments input.*/
  commentsInputLabel: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the date range picker. */
  dateRangePickerLocaleCode: PropTypes.string,
  /** The label for the departure date input.*/
  departureDateInputLabel: PropTypes.string,
  /** The label for the email input.*/
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The label for the guests input.*/
  guestsInputLabel: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input.*/
  nameInputLabel: PropTypes.string,
  /** The function called when the property dropdown is changed.
   *  @param {String}        name - The name of the property dropdown field.
   *  @param {String|Number} value - The value of the property dropdown after the change
   */
  onChangeProperty: PropTypes.func,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input.*/
  phoneInputLabel: PropTypes.string,
  /** The label for the property input.*/
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The indent level of a property option. One of: 0, 1, 2, 3, 4, 5 */
      indent: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The label for the room input.*/
  roomInputLabel: PropTypes.string,
  /** The options which the user can select for the room field. */
  roomOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The text to display on the submit button. */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    captcha: PropTypes.object,
    comments: PropTypes.object,
    dates: PropTypes.object,
    email: PropTypes.object,
    guests: PropTypes.object,
    name: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    room: PropTypes.object,
  }),
};
