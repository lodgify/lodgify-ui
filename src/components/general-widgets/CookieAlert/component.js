import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'elements/Modal';
import { FlexContainer } from 'layout/FlexContainer';
import { Button } from 'elements/Button';
import { Divider } from 'elements/Divider';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Paragraph } from 'typography/Paragraph';
import { withResponsive } from 'utils/with-responsive';
import { Link } from 'elements/Link';

import { getFormMarkup } from './utils/getFormMarkup';

const Component = ({
  buttonText,
  isOpen,
  isUserOnMobile,
  linkText,
  linkUrl,
  onAccept,
  text,
}) => {
  if (!isOpen) return false;
  return isUserOnMobile ? (
    <Modal hasCloseIcon={false} hasRoundedCorners isOpen>
      <HorizontalGutters>
        <FlexContainer
          alignItems="center"
          flexDirection="column"
          justifyContent="center"
        >
          <Divider />
          <Paragraph>{text}</Paragraph>
          <Link href={linkUrl}>{linkText}</Link>
          <Divider />
          <Button onClick={onAccept}>{buttonText}</Button> <Divider />
        </FlexContainer>
      </HorizontalGutters>
    </Modal>
  ) : (
    <div className="cookies-form">
      {getFormMarkup(buttonText, linkText, linkUrl, onAccept, text)}
    </div>
  );
};

Component.displayName = 'CookieAlert';

Component.defaultProps = {
  buttonText: 'Accept',
  isOpen: false,
  linkText: 'Learn More',
  linkUrl: '/',
};

Component.propTypes = {
  /** The text to display on the button inside the component. */
  buttonText: PropTypes.string,
  /** Is the modal open. */
  isOpen: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The text to display as a link. */
  linkText: PropTypes.string,
  /** The url attached to the link. */
  linkUrl: PropTypes.string,
  /** The on accept handler for the call to action. */
  onAccept: PropTypes.func.isRequired,
  /** The text to display in the component. */
  text: PropTypes.string.isRequired,
  /** The element to be clicked to display the modal. */
};

export const ComponentWithResponsive = withResponsive(Component);
