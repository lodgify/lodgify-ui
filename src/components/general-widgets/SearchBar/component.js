import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import getClassNames from 'classnames';
import { Form } from 'semantic-ui-react';
import { isEqual } from 'lodash';

import { CHECK_OUR_AVAILABILITY, SEARCH } from 'utils/default-strings';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { GridRow } from 'layout/GridRow';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { Button } from 'elements/Button';

import { getFormFieldMarkup } from './utils/getFormFieldMarkup';
import { getSearchBarModal } from './utils/getSearchBarModal';

/**
 * The standard widget for property search.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {};

  componentDidUpdate(previousProps, previousState) {
    !isEqual(previousState, this.state) && this.props.onChangeInput(this.state);
  }

  persistInputChange = (name, value) => {
    this.setState({ [name]: value });
  };

  handleSubmit = () => {
    this.props.onSubmit(this.state);
  };

  render = () => {
    const {
      className,
      isDisplayedAsModal,
      isFixed,
      summaryElement,
      willDropdownsOpenAbove,
    } = this.props;

    const searchBarAsModal =
      isDisplayedAsModal &&
      getSearchBarModal(this.props, this.handleSubmit, this.persistInputChange);

    if (isFixed) {
      return (
        <div
          className={getClassNames(className, 'search-bar', {
            'is-fixed': isFixed,
          })}
        >
          <HorizontalGutters>
            <Grid>
              <GridRow verticalAlign="middle">
                <GridColumn width={5}>{summaryElement}</GridColumn>
                <GridColumn floated="right" width={7}>
                  {isDisplayedAsModal ? (
                    searchBarAsModal
                  ) : (
                    <Form onSubmit={this.handleSubmit}>
                      <Form.Group>
                        {getFormFieldMarkup(
                          this.props,
                          this.persistInputChange,
                          false,
                          true
                        )}
                      </Form.Group>
                    </Form>
                  )}
                </GridColumn>
              </GridRow>
            </Grid>
          </HorizontalGutters>
        </div>
      );
    }

    if (isDisplayedAsModal) {
      return searchBarAsModal;
    }

    return (
      <div className={getClassNames(className, 'search-bar')}>
        <Form onSubmit={this.handleSubmit}>
          <Form.Group>
            {getFormFieldMarkup(
              this.props,
              this.persistInputChange,
              false,
              willDropdownsOpenAbove
            )}
          </Form.Group>
        </Form>
      </div>
    );
  };
}

Component.displayName = 'SearchBar';

Component.defaultProps = {
  className: null,
  dateRangePickerLocaleCode: undefined,
  getIsDayBlocked: Function.prototype,
  isDisplayedAsModal: false,
  isFixed: false,
  isShowingSummary: false,
  isModalOpen: undefined,
  locationOptions: null,
  modalHeadingText: CHECK_OUR_AVAILABILITY,
  modalSummaryElement: null,
  modalTrigger: <Icon name={ICON_NAMES.SEARCH} />,
  onChangeInput: Function.prototype,
  onCloseModal: Function.prototype,
  onSubmit: Function.prototype,
  searchButton: (
    <Button icon={ICON_NAMES.SEARCH} isPositionedRight isRounded>
      {SEARCH}
    </Button>
  ),
  summaryElement: null,
  willDropdownsOpenAbove: false,
};

Component.propTypes = {
  /**
   * Custom class name.
   * Provided by `ShowOn` so ignored in the styleguide.
   * @ignore
   */
  className: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the date range picker. */
  // eslint-disable-next-line react/no-unused-prop-types
  dateRangePickerLocaleCode: PropTypes.string,
  /**
   * A function called for each day to be displayed in the DateRangePicker. Returning true blocks that day in the date range picker.
   * @param   {Moment}  day - The day to test.
   * @returns {boolean}     - Is the day blocked.
   */
  // eslint-disable-next-line react/no-unused-prop-types
  getIsDayBlocked: PropTypes.func,
  /** The options which the user can select in the guests field. */
  // eslint-disable-next-line react/no-unused-prop-types
  guestsOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** Is the Search Bar displayed in a modal*/
  isDisplayedAsModal: PropTypes.bool,
  /** Is the Search Bar fixed to the bottom of the window */
  isFixed: PropTypes.bool,
  /** Is the modal open. Used when consuming `SearchBar` as a controlled component. */
  // eslint-disable-next-line react/no-unused-prop-types
  isModalOpen: PropTypes.bool,
  /** Is Search Bar showing the Property Summary info. */
  // eslint-disable-next-line react/no-unused-prop-types
  isShowingSummary: PropTypes.bool,
  /** The options which the user can select in the location field. */
  // eslint-disable-next-line react/no-unused-prop-types
  locationOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The indent level of a location option. One of: 0, 1, 2, 3, 4, 5 */
      indent: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The heading text to display in the modal */
  // eslint-disable-next-line react/no-unused-prop-types
  modalHeadingText: PropTypes.string,
  /** The summary element to display in the mobile modal  */
  // eslint-disable-next-line react/no-unused-prop-types
  modalSummaryElement: PropTypes.node,
  /** The element to be clicked to display the modal. */
  // eslint-disable-next-line react/no-unused-prop-types
  modalTrigger: PropTypes.node,
  /** A function called when a change in an input occurs.
   *  @param {Object} values - The values of the inputs in the search bar.
   *  @param {Object} values.dates
   *  @param {String} values.guests
   *  @param {String} values.location
   */
  onChangeInput: PropTypes.func,
  /** A function called when a close event happens on the modal. Used when consuming `SearchBar` as a controlled component. */
  // eslint-disable-next-line react/no-unused-prop-types
  onCloseModal: PropTypes.func,
  /** The function to call when the search bar is submitted.
   *  @param {Object} values - The values of the inputs in the search bar.
   *  @param {Object} values.dates
   *  @param {String} values.guests
   *  @param {String} values.location
   */
  onSubmit: PropTypes.func,
  /** The Search Button the Search Bar displays. */
  // eslint-disable-next-line react/no-unused-prop-types
  searchButton: PropTypes.node,
  /** The element to display in the fixed container */
  summaryElement: PropTypes.node,
  /** The dropdowns will open above the search bar. */
  willDropdownsOpenAbove: PropTypes.bool,
};
