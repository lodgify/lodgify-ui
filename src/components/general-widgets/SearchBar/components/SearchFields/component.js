import React from 'react';
import {
  string,
  func,
  arrayOf,
  shape,
  oneOf,
  oneOfType,
  bool,
  number,
  object,
  node,
} from 'prop-types';

import { Button } from 'elements/Button';
import { size } from 'utils/size';
import {
  CHECK_IN,
  CHECK_OUT,
  GUESTS,
  LOCATION,
  SEARCH,
} from 'utils/default-strings';
import { ICON_NAMES } from 'elements/Icon';
import { Dropdown } from 'inputs/Dropdown';
import { CounterDropdown } from 'inputs/CounterDropdown';
import { DateRangePicker } from 'inputs/DateRangePicker';

export const Component = ({
  dateRangePickerLocaleCode,
  datesCheckInLabel,
  datesCheckOutLabel,
  datesInputFocusedInput,
  datesInputOnFocusChange,
  datesInputValue,
  getIsDayBlocked,
  guestsInputLabel,
  guestsInputValue,
  guestsPopupId,
  isCalendarIconDisplayed,
  isDateRangePickerLoading,
  locationInputLabel,
  locationInputValue,
  locationOptions,
  maximumGuestsInputValue,
  onInputChange,
  searchButton,
  willLocationDropdownOpenAbove,
}) => (
  <div className="inputs-container">
    {!!size(locationOptions) && (
      <div className="input-container location-input-container">
        <Dropdown
          icon={ICON_NAMES.MAP_PIN}
          label={locationInputLabel}
          name="location"
          onChange={onInputChange}
          options={locationOptions}
          value={locationInputValue}
          willOpenAbove={willLocationDropdownOpenAbove}
        />
      </div>
    )}
    <div className="input-container dates-input-container">
      <DateRangePicker
        endDatePlaceholderText={datesCheckOutLabel}
        focusedInput={datesInputFocusedInput}
        getIsDayBlocked={getIsDayBlocked}
        isCalendarIconDisplayed={isCalendarIconDisplayed}
        isLoading={isDateRangePickerLoading}
        localeCode={dateRangePickerLocaleCode}
        name="dates"
        onChange={onInputChange}
        onFocusChange={datesInputOnFocusChange}
        startDatePlaceholderText={datesCheckInLabel}
        value={datesInputValue}
        willOpenAbove={willLocationDropdownOpenAbove}
      />
    </div>
    <div className="input-container guests-input-container">
      <CounterDropdown
        counterName="guests"
        counterValue={guestsInputValue}
        dropdownLabel={guestsInputLabel}
        maximumCounterValue={maximumGuestsInputValue}
        onChange={onInputChange}
        popupId={guestsPopupId}
      />
    </div>
    <div className="button-container">{searchButton}</div>
  </div>
);

Component.displayName = 'SearchFields';

Component.defaultProps = {
  dateRangePickerLocaleCode: undefined,
  datesCheckInLabel: CHECK_IN,
  datesCheckOutLabel: CHECK_OUT,
  datesInputFocusedInput: undefined,
  datesInputOnFocusChange: Function.prototype,
  datesInputValue: undefined,
  getIsDayBlocked: Function.prototype,
  guestsInputLabel: GUESTS,
  guestsInputValue: undefined,
  guestsPopupId: undefined,
  isCalendarIconDisplayed: undefined,
  isDateRangePickerLoading: undefined,
  locationInputLabel: LOCATION,
  locationInputValue: undefined,
  locationOptions: null,
  maximumGuestsInputValue: undefined,
  searchButton: (
    <Button icon={ICON_NAMES.SEARCH} isFormSubmit isRounded>
      {SEARCH}
    </Button>
  ),
  willLocationDropdownOpenAbove: false,
  onInputChange: Function.prototype,
};

Component.propTypes = {
  dateRangePickerLocaleCode: string,
  datesCheckInLabel: string,
  datesCheckOutLabel: string,
  datesInputFocusedInput: string,
  datesInputOnFocusChange: func,
  datesInputValue: shape({
    endDate: object,
    startDate: object,
  }),
  getIsDayBlocked: func,
  guestsInputLabel: string,
  guestsInputValue: oneOfType([string, number]),
  guestsPopupId: string,
  isCalendarIconDisplayed: bool,
  isDateRangePickerLoading: bool,
  locationInputLabel: string,
  locationInputValue: string,
  locationOptions: arrayOf(
    shape({
      /** The indent level of a location option. One of: 0, 1, 2, 3, 4, 5 */
      indent: oneOf([0, 1, 2, 3, 4, 5]),
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  maximumGuestsInputValue: number,
  onInputChange: func,
  searchButton: node,
  willLocationDropdownOpenAbove: bool,
};
