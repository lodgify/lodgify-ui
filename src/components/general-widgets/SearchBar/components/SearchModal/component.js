import React, { useState } from 'react';
import { string, bool, node, func } from 'prop-types';
import { Form } from 'semantic-ui-react';

import { Heading } from 'typography/Heading';
import { Modal } from 'elements/Modal';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { CHECK_OUR_AVAILABILITY } from 'utils/default-strings';

import { SearchFields } from '../SearchFields';

export const Component = ({
  modalSummaryElement,
  modalHeadingText,
  isModalOpen,
  searchButton,
  onSubmit,
  ...searchFieldsProps
}) => {
  const [isOpen, setIsOpen] = useState(isModalOpen);

  return (
    <Modal
      hasPadding
      header={
        modalSummaryElement ? (
          modalSummaryElement
        ) : (
          <Heading size="small">{modalHeadingText}</Heading>
        )
      }
      isFullscreen
      isOpen={isOpen}
      onClose={() => setIsOpen(false)}
      trigger={
        <div
          data-test-id="search-bar-search-modal-trigger"
          onClick={() => setIsOpen(true)}
        >
          {searchButton}
        </div>
      }
    >
      <div className="search-bar is-stackable">
        <HorizontalGutters>
          <Form onSubmit={onSubmit}>
            <SearchFields {...searchFieldsProps} />
          </Form>
        </HorizontalGutters>
      </div>
    </Modal>
  );
};

Component.displayName = 'SearchModal';
Component.defaultProps = {
  modalSummaryElement: null,
  modalHeadingText: CHECK_OUR_AVAILABILITY,
  onSubmit: Function.prototype,
  isModalOpen: false,
  searchButton: <div />,
};
Component.propTypes = {
  isModalOpen: bool,
  modalHeadingText: string,
  modalSummaryElement: node,
  onSubmit: func,
  searchButton: node,
};
