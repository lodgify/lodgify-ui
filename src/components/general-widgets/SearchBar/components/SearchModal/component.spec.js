import React from 'react';
import { mount } from 'enzyme';

import { Component as SearchModal } from './component';

const props = {
  isModalOpen: false,
  modalHeadingText: 'Hello world',
  modalSummaryElement: null,
  onCloseModal: Function.prototype,
  searchButton: <div />,
  handleSubmit: () => {},
  onInputChange: f => f,
};

const getModalMarkup = extraProps =>
  mount(<SearchModal {...props} {...extraProps} />);

describe('getSearchBarModal', () => {
  it('should return the right structure', () => {
    const actual = getModalMarkup();

    expect(actual).toMatchSnapshot();
  });

  describe('if `props.modalSummaryElement` is passed', () => {
    it('should return the right structure', () => {
      const actual = getModalMarkup({
        modalSummaryElement: <div />,
      });

      expect(actual).toMatchSnapshot();
    });
  });
});
