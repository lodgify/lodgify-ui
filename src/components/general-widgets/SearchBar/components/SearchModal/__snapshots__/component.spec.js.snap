// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSearchBarModal if \`props.modalSummaryElement\` is passed should return the right structure 1`] = `
<SearchModal
  handleSubmit={[Function]}
  isModalOpen={false}
  modalHeadingText="Hello world"
  modalSummaryElement={<div />}
  onCloseModal={[Function]}
  onInputChange={[Function]}
  searchButton={<div />}
>
  <Modal
    hasCloseIcon={true}
    hasPadding={true}
    hasRoundedCorners={false}
    header={<div />}
    isFullscreen={true}
    isOpen={false}
    onClose={[Function]}
    size="tiny"
    trigger={<div />}
  >
    <Modal
      centered={true}
      className="has-padding"
      closeIcon={
        <Icon
          className={null}
          color={null}
          hasBorder={false}
          isButton={false}
          isCircular={false}
          isColorInverted={false}
          isDisabled={false}
          isLabelLeft={false}
          labelText={null}
          labelWeight={null}
          name="close"
          path={null}
          size={null}
        />
      }
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer="inverted"
      eventPool="Modal"
      onClose={[Function]}
      open={false}
      size="fullscreen"
      trigger={<div />}
    >
      <Portal
        closeOnDocumentClick={false}
        closeOnEscape={true}
        eventPool="Modal"
        mountNode={<body />}
        onClose={[Function]}
        onMount={[Function]}
        onOpen={[Function]}
        onUnmount={[Function]}
        open={false}
        openOnTriggerClick={true}
        trigger={<div />}
      >
        <Ref
          innerRef={[Function]}
        >
          <RefFindNode
            innerRef={[Function]}
          >
            <div
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
          </RefFindNode>
        </Ref>
      </Portal>
    </Modal>
  </Modal>
</SearchModal>
`;

exports[`getSearchBarModal should return the right structure 1`] = `
<SearchModal
  handleSubmit={[Function]}
  isModalOpen={false}
  modalHeadingText="Hello world"
  modalSummaryElement={null}
  onCloseModal={[Function]}
  onInputChange={[Function]}
  searchButton={<div />}
>
  <Modal
    hasCloseIcon={true}
    hasPadding={true}
    hasRoundedCorners={false}
    header={
      <Heading
        className={null}
        hasMargin={true}
        isColorInverted={false}
        size="small"
        textAlign={null}
      >
        Hello world
      </Heading>
    }
    isFullscreen={true}
    isOpen={false}
    onClose={[Function]}
    size="tiny"
    trigger={<div />}
  >
    <Modal
      centered={true}
      className="has-padding"
      closeIcon={
        <Icon
          className={null}
          color={null}
          hasBorder={false}
          isButton={false}
          isCircular={false}
          isColorInverted={false}
          isDisabled={false}
          isLabelLeft={false}
          labelText={null}
          labelWeight={null}
          name="close"
          path={null}
          size={null}
        />
      }
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer="inverted"
      eventPool="Modal"
      onClose={[Function]}
      open={false}
      size="fullscreen"
      trigger={<div />}
    >
      <Portal
        closeOnDocumentClick={false}
        closeOnEscape={true}
        eventPool="Modal"
        mountNode={<body />}
        onClose={[Function]}
        onMount={[Function]}
        onOpen={[Function]}
        onUnmount={[Function]}
        open={false}
        openOnTriggerClick={true}
        trigger={<div />}
      >
        <Ref
          innerRef={[Function]}
        >
          <RefFindNode
            innerRef={[Function]}
          >
            <div
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            />
          </RefFindNode>
        </Ref>
      </Portal>
    </Modal>
  </Modal>
</SearchModal>
`;
